From 31d35409fccc075e35b1045fef5cc8418e0ab36a Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 16 Mar 2016 18:59:50 -0700
Subject: [PATCH] laird: wb45n: wb50n: Add internal pull up to debug uart RX
 pin

By adding the pull up configuration, we avoid the possibility of noise on
the pin interrupting us when we get to u-boot. This is only important for
a use-case where the customer leaves the pin unconnected, but customers
do report about a 1% interrupted boot in that particular use-case.

Bug: 9195
---
 board/at91sam9x5ek/at91sam9x5ek.c | 2 +-
 board/sama5d3xek/sama5d3xek.c     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/at91sam9x5ek/at91sam9x5ek.c b/board/at91sam9x5ek/at91sam9x5ek.c
index a026b92..0b7ee7b 100644
--- a/board/at91sam9x5ek/at91sam9x5ek.c
+++ b/board/at91sam9x5ek/at91sam9x5ek.c
@@ -53,7 +53,7 @@ static void at91_dbgu_hw_init(void)
 {
 	/* Configure DBGU pins */
 	const struct pio_desc dbgu_pins[] = {
-		{"RXD", AT91C_PIN_PA(9), 0, PIO_DEFAULT, PIO_PERIPH_A},
+		{"RXD", AT91C_PIN_PA(9), 0, PIO_PULLUP, PIO_PERIPH_A},
 		{"TXD", AT91C_PIN_PA(10), 0, PIO_DEFAULT, PIO_PERIPH_A},
 		{(char *)0, 0, 0, PIO_DEFAULT, PIO_PERIPH_A},
 	};
diff --git a/board/sama5d3xek/sama5d3xek.c b/board/sama5d3xek/sama5d3xek.c
index 6ecd778..669fe5f 100644
--- a/board/sama5d3xek/sama5d3xek.c
+++ b/board/sama5d3xek/sama5d3xek.c
@@ -65,7 +65,7 @@ static void at91_dbgu_hw_init(void)
 {
 	/* Configure DBGU pin */
 	const struct pio_desc dbgu_pins[] = {
-		{"RXD", AT91C_PIN_PB(30), 0, PIO_DEFAULT, PIO_PERIPH_A},
+		{"RXD", AT91C_PIN_PB(30), 0, PIO_PULLUP, PIO_PERIPH_A},
 		{"TXD", AT91C_PIN_PB(31), 0, PIO_DEFAULT, PIO_PERIPH_A},
 		{(char *)0, 0, 0, PIO_DEFAULT, PIO_PERIPH_A},
 	};
-- 
1.9.1

